### ISSUE TITLE
[macOS 14] Erlang 26.1.1 Fails To Build With wxWidgets 3.2.2.1

### ISSUE THREAD

**Issue by conradwt** [erlang/otp#7696]
**Describe the bug**

Erlang/OTP 26.1.1 Fails To Build With wxWidgets 3.2.2.1

**To Reproduce**

1. build wxWidgets

```zsh
git clone git@github.com:wxWidgets/wxWidgets.git
cd wxWidgets
git co tags/v3.2.2.1
git submodule update --init src/png
git submodule update --init src/jpeg
git submodule update --init src/zlib
git submodule update --init src/tiff
git submodule update --init src/expat
git submodule update --init 3rdparty/pcre
git submodule update --init 3rdparty/catch
git submodule update --init 3rdparty/nanosvg
./configure \
  --with-cocoa \
  --enable-webview \
  --enable-compat28 \
  --prefix=/usr/local \
  --with-expat=builtin \
  --with-libjpeg=builtin \
  --with-libpng=builtin \
  --with-libtiff=builtin \
  --with-regex=builtin \
  --with-macosx-version-min=${wxUSE_MACOSX_VERSION_MIN} \
  --with-zlib=builtin
make
sudo make install
```

2. build Erlang/OTP 26.1.1

```zsh
export CC=clang
export CPP="clang -E"
export EGREP=egrep
export FOP_HOME=/opt/local/bin
export KERL_BUILD_DOCS=yes
export KERL_INSTALL_MANPAGES=yes
export KERL_USE_AUTOCONF=0
export wxUSE_MACOSX_VERSION_MIN=${$(sw_vers -productVersion)%.*}
export KERL_CONFIGURE_OPTIONS="--disable-debug \
                               --disable-silent-rules \
                               --enable-dynamic-ssl-lib \
                               --enable-gettimeofday-as-os-system-time \
                               --enable-kernel-poll \
                               --with-ssl=/opt/local \
                               --without-javac \
                               --without-odbc"
asdf install erlang 26.1.1
```

**Expected behavior**

I expected it to build without error.

**Affected versions**

Erlang/OTP 26.1.1

**Additional context**

Development Setup:

OS: macOS 14 (Sonoma)
Xcode: Xcode Version 15.0 (15A240d)
Machine: Apple Silicon M1 Max MacBook Pro

Here's a partial error log.

```
1 warning and 1 error generated.
make[3]: *** [aarch64-apple-darwin23.0.0/wxe_wrapper_7.o] Error 1
make[3]: *** Waiting for unfinished jobs....
1 warning generated.
1 warning generated.
1 warning generated.
1 warning generated.
1 warning generated.
1 warning generated.
1 warning generated.
1 warning generated.
5 warnings generated.
make[2]: *** [opt] Error 2
make[1]: *** [opt] Error 2
make: *** [libs] Error 2
```

With Erlang/OTP 26.1, it builds without error and generates the following warning for wx 3.1.6 and 3.2.2.1:

```zsh
 * wx             : Can not link wx program are all developer packages installed?
 ```

**Comment by dgud** [comment]
Remove the --enable-compat28  when building wxWidgets it should not be there.
The build instructions was for wxWidgets-3.0  and older OTP versions (Björn have update them in the latest release).

**Comment by mangalakader** [comment]
For some reason, I had the same issue as @conradwt , I was trying to compile `wxWidgets` from source using various options and then installing Erlang using `asdf` with errors.

After doing a clean uninstall of `wxWidgets` using the following commands:

```
cd wxWidgets
sudo make uninstall
make distclean
make clean

# The binaries are not automatically cleaned up

rm -f  /usr/bin/wx*
rm -r  /usr/include/wx*
rm -r  /usr/lib/wx
rm -f  /usr/lib/libwx*
rm -f  /usr/local/bin/wx*
rm -r  /usr/local/include/wx*
rm -r  /usr/local/lib/wx
rm -f  /usr/local/lib/libwx*

# The above is a dirty way to force clean the binaries
```
You do a install of `wxWidgets` using brew which installs the version 3.2.2.1

And I had the following configuration in my rc file:
```
KERL_CONFIGURE_OPTIONS="--disable-debug --disable-silent-rules --enable-hipe --enable-darwin-64bit --enable-kernel-poll --enable-shared-zlib --enable-smp-support --enable-threads --enable-dynamic-ssl-lib --enable-webview --enable-wx --with-odbc=/opt/homebrew/opt/unixodbc --with-wx --with-wx-config=/opt/homebrew/bin/wx-config --with-ssl=/opt/homebrew/opt/openssl@3"
KERL_BUILD_DOCS=yes
KERL_INSTALL_MANPAGES=yes
KERL_USE_AUTOCONF=0
LDFLAGS="${LDFLAGS+"$LDFLAGS "}-L/opt/homebrew/lib -L/opt/homebrew/opt/libpq/lib -L/opt/homebrew/opt/unixodbc/lib -L/opt/homebrew/opt/openssl@3/lib -L/usr/local/lib"
CPPFLAGS="${CPPFLAGS+"$CPPFLAGS "}-I/opt/homebrew/include -I/opt/homebrew/opt/libpq/include -I/opt/homebrew/opt/unixodbc/include -I/opt/homebrew/opt/openssl@3/include -I/usr/local/include"
```

Open a new terminal window and install Erlang (v26.1.1) using asdf as usual and it worked.

After installation, I got the observer to be able to start.....

**Comment by conradwt** [comment]
@dgud Thanks for that because it has resolved the build issues for Erlang/OTP 26.1.1 for me.

**Comment by dgud** [comment]
I could reproduce on linux, so was easy this time.
Closing this.
